R/update.R

```diff
@@ -77,6 +83,62 @@ renv_update_find_github_impl <- function(record) {
 
 }
 
+renv_update_find_git_impl <- function(record) {
+
+  # validate we have a ref
+  if (!renv_record_validate(record))
+    return(NULL)
+
+  renv_git_preflight()
```

Comment:
Looks like a lot of this was copied from here:

https://github.com/rstudio/renv/blob/25ed22481bdeb89b5dde5f9ab5bc971597b14c4c/R/retrieve.R#L258-L298

Could we refactor that part into a function `renv_retrieve_git_impl(record)`, and then use that both here and in `renv_retrieve_git()`?

Reply:
I finally came back to resolve this, but now I am confused about the intended functionality. 

If an R package is installed from a feature branch, then it seems logical that the update process will only check the feature branch for updates. Similarly, if a package is installed form main branch, then only the main branch will be checked for updates.

If an R package is installed based on a commit ID, then it seems logical that no updates will ever occur because a commit ID will not change. However, the user may intend to install based on a commit ID and then receive updates based on the corresponding branch. 

I am not sure how to code for the different possibilities of what parameters were used to install and whether or not it is possible/logical to perform updates.

Reply:
Right now, `renv::restore()` always chooses to restore the exact commit associated with whatever reference was supplied. We might change this in the future (likely via an option / opt-in mechanism) but that is currently intentional for both cases.